File: TapeEquilibrium/tape_equilibrium.cpp
#include "tape_equilibrium.h"

int tape_equilibrium::solution(std::vector<int>& A)
{
    if (A.empty())
        return 0;

    int right = 0;
    int left = A[0];
    const int size = A.size();

    for (int i = 1; i < size; ++i)
        right += A[i];

    int ans = abs(left - right);
    
    for (int i = 1; i < size; ++i)
    {
        ans = abs(left - right) < ans ? abs(left - right) : ans;
        left += A[i];
        right -= A[i];
    }

    return ans;
}

File: csharp/TapeEquilibrium/TapeEquilibrium.cs
using System;

public static class TapeEquilibrium
{
    public static int Solution(int[] a)
    {
        if (a.Length <= 0)
            return 0;

        var left = a[0];
        var right = 0;

        for (int i = 1; i < a.Length; i++)
            right += a[i];

        var result = Math.Abs(right - left);
        
        for (int i = 1; i < a.Length; i++)
        {
            if (Math.Abs(right - left) < result)
                result = Math.Abs(right - left);
            
            left += a[i];
            right -= a[i];
        }

        return result;
    }
}